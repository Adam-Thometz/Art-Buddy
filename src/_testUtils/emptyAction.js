const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION