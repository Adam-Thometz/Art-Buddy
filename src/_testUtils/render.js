import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
