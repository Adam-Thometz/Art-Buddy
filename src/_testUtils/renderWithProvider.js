import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;