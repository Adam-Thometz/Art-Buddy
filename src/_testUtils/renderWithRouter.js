import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;