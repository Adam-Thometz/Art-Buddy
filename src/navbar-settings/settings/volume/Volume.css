.Volume {
  display: flex;
  justify-content: space-between;
}