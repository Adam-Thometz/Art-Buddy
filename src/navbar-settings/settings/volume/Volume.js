import { useSelector, useDispatch } from "react-redux";
import { setVolume } from "_redux/settings/settingsReducer";

import "./Volume.css";

import Icon from "_components/icon/Icon";

import { VOLUME_LOW, VOLUME_MEDIUM, VOLUME_HIGH } from "_media/settings/settings.assets";
import { LOW, MEDIUM, HIGH } from "_data/_utils/volume";

const Volume = () => {
  const { volume } = useSelector((state) => state.settings);
  const dispatch = useDispatch();

  const selectedVolume = {
    // Space Cadet (not completely accurate)
    filter:
      "invert(9%) sepia(31%) saturate(5353%) hue-rotate(221deg) brightness(100%) contrast(94%)",
  };

  const unselectedVolume = {
    // Light Grey (not completely accurate)
    filter:
      "invert(71%) sepia(83%) saturate(6%) hue-rotate(236deg) brightness(103%) contrast(89%)",
  };

  const handleVolume = (e) => dispatch(setVolume(+e.currentTarget.id));

  return (
    <section className="Volume">
      <Icon
        icon={VOLUME_LOW}
        otherImgStyles={volume === LOW ? selectedVolume : unselectedVolume}
        text="LOW"
        onClick={handleVolume}
        id={LOW}
        size="50px"
        width="40px"
      />
      <Icon
        icon={VOLUME_MEDIUM}
        otherImgStyles={volume === MEDIUM ? selectedVolume : unselectedVolume}
        text="MEDIUM"
        onClick={handleVolume}
        id={MEDIUM}
        size="50px"
      />
      <Icon
        icon={VOLUME_HIGH}
        otherImgStyles={volume === HIGH ? selectedVolume : unselectedVolume}
        text="HIGH"
        onClick={handleVolume}
        id={HIGH}
        size="50px"
        width="60px"
      />
    </section>
  );
};

export default Volume;
