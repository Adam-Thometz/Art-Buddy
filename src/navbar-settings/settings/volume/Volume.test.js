import Volume from './Volume';

import { render } from "_testUtils/render";

describe('Volume component', () => {
  it('renders without crashing', () => {
    render(<Volume />);
  });

  it('matches the snapshot of Volume', () => {
    const { asFragment } = render(<Volume />);
    expect(asFragment()).toMatchSnapshot();
  });
});