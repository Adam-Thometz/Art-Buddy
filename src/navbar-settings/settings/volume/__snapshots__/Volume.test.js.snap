// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Volume component matches the snapshot of Volume 1`] = `
<DocumentFragment>
  <section
    class="Volume"
  >
    <figure
      class="Icon"
      data-testid=""
      id="-12"
    >
      <img
        alt=""
        src="https://art-buddy.s3.amazonaws.com/settings/volume/volume-low.png"
        style="height: 50px; width: 40px; cursor: pointer; filter: invert(71%) sepia(83%) saturate(6%) hue-rotate(236deg) brightness(103%) contrast(89%);"
      />
      <figcaption
        class="Icon-text"
        style="font-size: 1rem;"
      >
        LOW
      </figcaption>
    </figure>
    <figure
      class="Icon"
      data-testid=""
      id="0"
    >
      <img
        alt=""
        src="https://art-buddy.s3.amazonaws.com/settings/volume/volume-medium.png"
        style="height: 50px; width: 50px; cursor: pointer; filter: invert(9%) sepia(31%) saturate(5353%) hue-rotate(221deg) brightness(100%) contrast(94%);"
      />
      <figcaption
        class="Icon-text"
        style="font-size: 1rem;"
      >
        MEDIUM
      </figcaption>
    </figure>
    <figure
      class="Icon"
      data-testid=""
      id="12"
    >
      <img
        alt=""
        src="https://art-buddy.s3.amazonaws.com/settings/volume/volume-high.png"
        style="height: 50px; width: 60px; cursor: pointer; filter: invert(71%) sepia(83%) saturate(6%) hue-rotate(236deg) brightness(103%) contrast(89%);"
      />
      <figcaption
        class="Icon-text"
        style="font-size: 1rem;"
      >
        HIGH
      </figcaption>
    </figure>
  </section>
</DocumentFragment>
`;
